shared/src/main/scala/org/adridadou/openlaw/parser/template/variableTypes/VariableType.scala

```diff
@@ -127,14 +127,17 @@ abstract class VariableType(val name: String) {
 
   def internalFormat(value: Any): String
 
-  def construct(constructorParams: Parameter, executionResult: TemplateExecutionResult): Option[Any] = constructorParams match {
+  def construct(constructorParams: Parameter, executionResult: TemplateExecutionResult): Either[Throwable, Option[Any]] = constructorParams match {
       case OneValueParameter(expr) =>
-        expr.evaluate(executionResult)
+        Try(expr.evaluate(executionResult)).toEither
       case Parameters(parameterMap) =>
-        parameterMap.toMap.get("value")
-          .flatMap(construct(_, executionResult))
+        parameterMap.toMap.get("value") match {
```

Comment:
why would it be simpler? I know that parameterMap.toMap.get(...) will never throw but construct might. So I put the Try around the relevant code

Reply:
Simpler because you don't have to mess around with manually converting the monad in the matcher.  But I understand the point about only wrapping the minimum amount of code in the Try.  It could still likely be done with built in monad transformations, but this is largely a style issue.

The code is equivalent, but I find manually manipulating the monads to be less clear and more error prone; merely a personal opinion.  That's why I noted it as a NIT and not a request for a change. :)